src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+    var Sequence_With_Y_Position: List<String> = emptyList()
+    var dataValues = ArrayList<String>()
+    var tableColumnOrderArrayID = ArrayList<String>()
+    val tableColumnType = mutableListOf<String>()
+    val tableColumnTitle = mutableListOf<String>()
+    val tableColumnValue = mutableListOf<String>()
+    var type: String? = null
+    var list= ArrayList<componentModel>()
+    val mainHandler = Handler(context.mainLooper)
+    var options: JSONArray? = null
+    private val client = OkHttpClient()
+    object SharedData {
+        var joyfillDocID: String = ""
+        var authorizedToken: String = ""
+    }
+    val textField = TextField(context)
+    val dropDown = DropDown(context)
+    val multipleChoice = MultipleChoice(context)
+
+
+    val typePositionsMap = mutableMapOf<String, Int>()
+    val idPositionsMap = mutableMapOf<String, Int>()
+
+    init {
+        orientation = VERTICAL
+        val layoutParams = LayoutParams(
+            LayoutParams.MATCH_PARENT,
+            LayoutParams.MATCH_PARENT
+        )
+        setLayoutParams(layoutParams)
+
+        layoutParams.setMargins(dpToPx(context, 8), dpToPx(context, 8), dpToPx(context, 8), dpToPx(context, 8))
+        val componentLayoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        componentLayout.orientation = LinearLayout.VERTICAL
+        componentLayout.layoutParams = componentLayoutParams
+
+        val view = LinearLayout(context)
+        val viewLayoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT,
+        )
+
+        view.layoutParams = viewLayoutParams
+        view.orientation = VERTICAL
+        view.background = layoutBackground()
+
+        val scrollView = NestedScrollView(context)
+        val scrollViewParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+        scrollViewParams.setMargins(0, dpToPx(context, 10), 0, dpToPx(context, 10))
+        scrollView.layoutParams= scrollViewParams
+        scrollView.setVerticalScrollBarEnabled(false)
+        scrollView.overScrollMode= View.OVER_SCROLL_NEVER
+        scrollView.fullScroll(ScrollView.FOCUS_DOWN)
+
+        scrollView.addView(componentLayout)
+        view.addView(scrollView)
+        addView(view)
+        makeApiCall()
+    }
+
+    //Mark: Function for API Call
+    fun makeApiCall() {
+        val url = "https://api-joy.joyfill.io/v1/documents/"+ SharedData.joyfillDocID
+        val request = Request.Builder()
+            .url(url)
+            .addHeader("Authorization", SharedData.authorizedToken)
+            .addHeader("Content-Type", "application/json")
+            .build()
+
+        client.newCall(request).enqueue(object : okhttp3.Callback {
```

Comment:
So 2 things here:
1. Why are we making an individual call in a TableField? Wouldn't the the data all be passed into this field from the developers making the call themselves?
2. If we do have to make an API call we would want to make it reusable if possible and only have added stuff in here that is specific to the TableField. For instance the request builder stuff and api-joy would want to make all reusable so you can easily put it into any component moving forward.

I may be confused on why you are doing this here so let me know.

Reply:
1. This class is not an individual call for Table Field. It is global for all the components.
2. Class MobileSDKField is reusable for all the components. It is a standard way to implement API